### PR TITLE
Subscription settings: remove extraneus "support info" bubble

### PR DESCRIPTION
**Pull request by simison**
Before

image

After

Screenshot 2024-05-08 at 14 51 50

## Proposed changes:

* Remove extra support link + privacy link from modal setting, as it seemed so disattached
* Adds descriptive line under "subscriptions" settings section.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

### CODE REVIEW

**Comment by crisbusquets** [comment]
Works for me!

@simison regarding what you mentioned on Slack (p1715169375392699/1715168088.134849-slack-C052XEUUBL4), I'd use `subscription forms`. `Newsletter` is the product itself, and visitors use subscribe forms to subscribe.

File: projects/plugins/jetpack/_inc/client/newsletter/subscriptions-settings.jsx [hunk 1]
```diff
@@ -101,7 +101,13 @@ function SubscriptionsSettings( props ) {
 			module={ SUBSCRIPTIONS_MODULE_NAME }
 			header={ __( 'Subscriptions', 'jetpack' ) }
 		>
-			<SettingsGroup disableInOfflineMode disableInSiteConnectionMode module={ subscriptions }>
+			<SettingsGroup disableInOfflineMode disableInSiteConnectionMode module={ subscriptions } >
```
Comment (pkuliga) [inline projects/plugins/jetpack/_inc/client/newsletter/subscriptions-settings.jsx]: Let's remove the trailing space.
```suggestion
			<SettingsGroup disableInOfflineMode disableInSiteConnectionMode module={ subscriptions }>
```

**Comment by pkuliga** [comment]
And let's remove not used `getRedirectUrl` and `SupportInfo` from the imports section.

**Comment by simison** [comment]
Lints fixed

**Review by pkuliga — APPROVED** [review]
Looks good now, thanks!

**Pull request merged**